app/lib/supabase/supabase-realtime.ts

```diff
@@ -1,352 +1,109 @@
-  typeof document !== 'undefined' && !document.hidden;
-
-const maxRetries = 3;
-
-/**
- * Subscribes to a Supabase Realtime channel when the component mounts and unsubscribes when the component unmounts.
- * Manages channel reconnection in case of errors and timeouts which can be caused by the tab going to the background, network connection issues, phone pausing the
- * execution of the app when in background, etc.
- *
- * @description
- * Hook's lifecycle starts in the 'subscribing' status and subscription is triggered on mount. When the hook is unmounted, the subscription is unsubscribed.
- *
- * 1. The hook listens to the changes of the channel status and acts accordingly:
- * - If the status is 'CLOSED', the hook unsubscribes from the channel.
- * - If the status is 'CHANNEL_ERROR' or 'TIMED_OUT':
- *   - If the tab is visible and the app is online, the hook retries the subscription (up to {@link maxRetries} times). During the retries the hook status is set to 'reconnecting'. If all the
- *     retries fail, the hook status is set to 'error' and the hook throws the error which is then caught by the error boundary.
- *   - If the tab is not visible (in the background) or the app is offline, the hook unsubscribes from the channel, which results in channel being closed and hook status being
- *     set to 'closed'.
- * - If the status is 'SUBSCRIBED', the hook does nothing and waits for the system postgres_changes ok message to be received (see https://github.com/supabase/realtime/issues/282
- *   for explanation and {@link setupSystemMessageListener} for implementation). Only when this message is received, the postgres_changes subscription is fully established, so
- *   the hook state is set to 'subscribed'. Every time when the system postgres_changes ok message is received (after the initial subscription or after resubscription), the hook
- *   calls the {@link onConnected} callback.
- *
- * 2. The hook listens for the visibility change of the tab and resubscribes to the channel if the tab is visible, the app is online and the channel is not already in 'joined'
- *    or 'joining' state. This makes sure that if the channel was closed while in background (either by our error/timeout handling or by the browser/machine), it will be
- *    reconnected when the tab is visible again.
- *
- * 3. The hook listens for the online status of the browser and resubscribes to the channel if the app comes back online, the tab is visible and the channel is not already
- *    in 'joined' or 'joining' state. This makes sure that if the channel was closed due to the lost network connection, it will be reconnected when the app comes back online.
- *
- * @param options - Subscription configuration.
- * @returns The status of the subscription.
- * @throws {Error} If the subscription errors while the app is in the foreground and all the retries fail.
- */
-export function useSupabaseRealtimeSubscription({
+export function useSupabaseRealtime({
```

Comment:
I don't like the API of this hook that much. I think it can be simplified but will need to play more with it. I didn't want to stall this PR even more